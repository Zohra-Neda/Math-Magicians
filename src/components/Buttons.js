import React from 'react';
import PropTypes from 'prop-types';

const Buttons = ({ handleButtonClick }) => (
  <div className="buttons">
    <button type="button" className="btn gray" onClick={handleButtonClick}>AC</button>
    <button type="button" className="btn gray" onClick={handleButtonClick}>+/-</button>
    <button type="button" className="btn gray" onClick={handleButtonClick}>%</button>
    <button type="button" className="btn orange" onClick={handleButtonClick}>รท</button>
    <button type="button" className="btn gray" onClick={handleButtonClick}>7</button>
    <button type="button" className="btn gray" onClick={handleButtonClick}>8</button>
    <button type="button" className="btn gray" onClick={handleButtonClick}>9</button>
    <button type="button" className="btn orange" onClick={handleButtonClick}>x</button>
    <button type="button" className="btn gray" onClick={handleButtonClick}>4</button>
    <button type="button" className="btn gray" onClick={handleButtonClick}>5</button>
    <button type="button" className="btn gray" onClick={handleButtonClick}>6</button>
    <button type="button" className="btn orange" onClick={handleButtonClick}>-</button>
    <button type="button" className="btn gray" onClick={handleButtonClick}>1</button>
    <button type="button" className="btn gray" onClick={handleButtonClick}>2</button>
    <button type="button" className="btn gray" onClick={handleButtonClick}>3</button>
    <button type="button" className="btn orange" onClick={handleButtonClick}>+</button>
    <button type="button" className="btn gray span" onClick={handleButtonClick}>0</button>
    <button type="button" className="btn gray" onClick={handleButtonClick}>.</button>
    <button type="button" className="btn orange" onClick={handleButtonClick}>=</button>
  </div>
);

Buttons.propTypes = {
  handleButtonClick: PropTypes.func.isRequired,
};
export default Buttons;
