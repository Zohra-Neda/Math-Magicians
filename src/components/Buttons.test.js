import React from 'react';
import { render } from '@testing-library/react';
import Buttons from './Buttons';

describe('Buttons', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<Buttons handleButtonClick={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
