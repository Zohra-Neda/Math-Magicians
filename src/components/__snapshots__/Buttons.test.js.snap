// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="buttons"
  >
    <button
      class="btn gray"
      type="button"
    >
      AC
    </button>
    <button
      class="btn gray"
      type="button"
    >
      +/-
    </button>
    <button
      class="btn gray"
      type="button"
    >
      %
    </button>
    <button
      class="btn orange"
      type="button"
    >
      รท
    </button>
    <button
      class="btn gray"
      type="button"
    >
      7
    </button>
    <button
      class="btn gray"
      type="button"
    >
      8
    </button>
    <button
      class="btn gray"
      type="button"
    >
      9
    </button>
    <button
      class="btn orange"
      type="button"
    >
      x
    </button>
    <button
      class="btn gray"
      type="button"
    >
      4
    </button>
    <button
      class="btn gray"
      type="button"
    >
      5
    </button>
    <button
      class="btn gray"
      type="button"
    >
      6
    </button>
    <button
      class="btn orange"
      type="button"
    >
      -
    </button>
    <button
      class="btn gray"
      type="button"
    >
      1
    </button>
    <button
      class="btn gray"
      type="button"
    >
      2
    </button>
    <button
      class="btn gray"
      type="button"
    >
      3
    </button>
    <button
      class="btn orange"
      type="button"
    >
      +
    </button>
    <button
      class="btn gray span"
      type="button"
    >
      0
    </button>
    <button
      class="btn gray"
      type="button"
    >
      .
    </button>
    <button
      class="btn orange"
      type="button"
    >
      =
    </button>
  </div>
</DocumentFragment>
`;
